chain/jsonrpc/src/lib.rs

```diff
@@ -1358,10 +1359,17 @@ impl JsonRpcHandler {
 fn rpc_handler(
     message: web::Json<Message>,
     handler: web::Data<JsonRpcHandler>,
-) -> impl Future<Output = Result<HttpResponse, HttpError>> {
+) -> impl Future<Output = HttpResponse> {
```

Comment:
To me it seems like we would never really return a HttpError anyway as `handler.process` never returns an error.

Sorry, could you help me understand the high level motive here?

Reply:
Oh I see, this is supposed to fix the original issue by Bowen. I think what we might want to do here is return the `Response` from `handler.process` and parse the response `result` field to see whether it's a `Value` or `RpcError`. In case of RpcError, convert to HttpError and return that?